anvio/utils.py

```diff
@@ -2362,17 +2362,34 @@ def export_sequences_from_contigs_db(contigs_db_path, output_file_path, seq_name
         else:
             seq_names_to_export = sorted(contig_sequences_dict.keys())
     else:
+        contig_names = [contig_name for contig_name in seq_names_to_export if contig_name in contig_sequences_dict]
+        split_names = [split_name for split_name in seq_names_to_export if split_name in splits_info_dict]
+        missing_names = [name for name in seq_names_to_export if name not in contig_names and name not in split_names]
+
         if splits_mode:
-            pass
-        else:
-            missing_names = [contig_name for contig_name in seq_names_to_export if contig_name not in contig_sequences_dict]
-            if len(missing_names):
-                if len(missing_names) == len(seq_names_to_export):
-                    raise ConfigError("None of the sequence names you have requested were found in this contigs database :/ "
-                                      "This is kind of impressive.")
+           if len(contig_names) or len(missing_names):
+                if just_do_it:
+                    run.warning("You are in splits mode, but not all the names you requested are splits in this CONTIGS.db. %d names are splits, "
+                                "%d are contigs, and %d are neither (of the %d total names that you gave). BUT you're in just-do-it mode and we know you're in charge, so we'll "
+                                "proceed using any recognizable split names." % (len(split_names), len(contig_names), len(missing_names),len(seq_names_to_export)))
+                    seq_names_to_export = split_names
                 else:
-                    raise ConfigError("The sequence names you have requested include those that are not in the contigs "
-                                      "database, which correspond to %d of %d names you've sent." % (len(missing_names), len(seq_names_to_export)))
+                    raise ConfigError("You are in splits mode, but not all the names you requested are splits in this CONTIGS.db. %d names are splits, "
+                                  "%d are contigs, and %d are neither (of the %d total names that you gave). If you want to live on the edge and try to "
+                                  "proceed using any recognizable split names, try out the `--just-do-it` flag." % (len(split_names), len(contig_names), len(missing_names),\
+                                  len(seq_names_to_export)))
```

Comment:
some of these lines are not aligned to the first line here. and instead of this:

```
(...)
    "proceed using any recognizable split names, try out the `--just-do-it` flag." % (len(split_names), len(contig_names), len(missing_names),\
    len(seq_names_to_export)))
(...)
```

You could do this for readability:

```
(...)
    "proceed using any recognizable split names, try out the `--just-do-it` flag." % \
    (len(split_names), len(contig_names), len(missing_names), len(seq_names_to_export)))
(...)
```